test/http-client.spec.js

```diff
@@ -527,6 +527,98 @@ describe('HttpClient', () => {
         });
     });
   });
+
+  describe('retry', () => {
```

Comment:
The more I think about it, the less i think this requirement makes sense here. I think the documentation should mention that the retry interceptor should be the last interceptor added to the interceptor collection. I mean, by the very fact that I'm re-calling `fetch` in the `responseError` callback, it will definitely mess up things for other interceptors, so it should be the last thing in the list. 

Thoughts?

Reply:
@davismj check the code added in the latest commit: https://github.com/aurelia/fetch-client/pull/104/commits/413fd4861a2cce0d8e6ea2201c2a8873427aa2e2#diff-ab4b11a4a0f024bbfafeda47a386d545R87

Reply:
It's hard to say. I could imagine a case where a retry interceptor would want to come first, e.g. when the interceptor wants to transform and normalize the error response, or where the retry interceptor would come last, e.g. when the interceptor wants to check a local cache to see if there is a recent copy of the record available.

The feature still works well without this flexibility, but I imagine it's might stop someone someday. I'm for kicking that can down the road.